addon/components/horizontal-bar-chart.js

```diff
@@ -306,7 +311,15 @@ export default ChartComponent.extend(FloatingTooltipMixin,
       .text((d) => this.get('formatLabelFunction')(d.value))
       .attr(this.get('valueLabelAttrs'));
 
-    var labelWidth = this.get('labelWidth');
+    var labelWidth;
+    // If the chart contains a mix of negative and positive values, the axis
+    // and labels are in the middle of the chart, not at the edges of the chart,
+    // so allow more space to compute how the label is trimmed.
+    if (this.get('hasNegativeValues') && this.get('hasPositiveValues')) {
+      labelWidth = this.get('outerWidth') / 2;
+    } else {
+      labelWidth = this.get('labelWidth');
+    }
```

Comment:
@akhileshgupta the slider still provides good value. I agree that the name of the slider is a misnomer, but it shouldn't warrant removal. We could change the name specifically for that context. It's not the optimal, but after playing around with the slider, the effect should be pretty obvious